src/data/types.rs

```diff
@@ -386,3 +392,37 @@ impl FunctionType {
         *self.return_type != Type::Void
     }
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use proptest::prelude::*;
+
+    use super::Type;
+
+    pub(crate) fn arb_type() -> impl Strategy<Value = Type> {
+        let leaf = prop_oneof![
+            Just(Type::Void),
+            Just(Type::Bool),
+            any::<bool>().prop_map(Type::Char),
+            any::<bool>().prop_map(Type::Short),
+            any::<bool>().prop_map(Type::Int),
+            any::<bool>().prop_map(Type::Long),
+            Just(Type::Float),
+            Just(Type::Double),
+            //Type::Enum(Option<InternedStr>, Vec<(InternedStr, i64)>),
+            Just(Type::VaList),
+            Just(Type::Error),
+        ];
+
+        leaf.prop_recursive(8, 256, 10, |inner| {
+            prop_oneof![
+                inner.prop_map(|t| Type::Pointer(Box::new(t))),
+                //Type::Function(FunctionType),
+                //Type::Union(StructType),
+                //Type::Struct(StructType),
+                //Type::Array(Box<Type>, ArrayType),
+                //Type::Bitfield(Vec<BitfieldType>),
```

Comment:
This can be removed, if/when I implement bitfields they will be annotations on StructType, not a separate variant.